src/modules/emote_menu/stores/twitch-emotes.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable import/prefer-default-export */
+import twitchApi from '../../../utils/twitch-api.js';
+import Emote from '../../emotes/emote.js';
+import Icons from '../components/Icons.jsx';
+import debug from '../../../utils/debug.js';
+import {getEmoteFromRegEx} from '../../../utils/regex.js';
+
+const EMOTE_SET_QUERY = `
+query UserEmotes {
+  currentUser {
+    emoteSets {
+      emotes {
+        id,
+        token
+      },
+      id,
+      owner {
+        id,
+        displayName
+        profileImageURL(width: 300)
+      }
+    }
+  }
+}
+`;
+
+const TWITCH_EMOTE_CDN = (id, size) => `https://static-cdn.jtvnw.net/emoticons/v2/${id}/default/dark/${size}`;
+
+function getForcedProviderToChannels(key) {
+  switch (key) {
+    case 0:
+      return {
+        id: 0,
```

Comment:
is there a way to avoid using a union type here? can id be only a string or an integer?